### DISCUSSION TITLE
Is it available to use Rc<RefCell<>> in spawned tasks

### DISCUSSION THREAD

**Discussion by ActivePeter** [tokio-rs/tokio#4371] [Q&A] [answered]
a

**Comment by Darksonn** [comment, marked as answer]
In general, a Tokio task may be moved from one thread to another at any `.await`, but it is not safe to move an `Rc` between threads. So to use an `Rc` in a spawned task, you must do one of the following:

 1. Make sure that the `Rc` is sufficiently short-lived that it does not have to exist when you perform an `.await`.
 2. Spawn your task on a `LocalSet` so that it can't be moved between threads.
